networkx/drawing/layout.py

```diff
@@ -894,6 +899,32 @@ def rescale_layout(pos, scale=1):
             pos[:, i] *= scale / lim
     return pos
 
+def rescale_layout_dict(pos, scale=1):
+    """Return a dictionary of scaled positions keyed by node 
+
+    Parameters
+    ----------
+    pos : A dictionary of positions keyed by node
+
+    scale : number (default: 1)
+        The size of the resulting extent in all directions.
+
+    Returns
+    -------
+    pos : A dictionary of positions keyed by node
+
+    See Also
+    --------
+    rescale_layout
+    """
+    import numpy as np
+    
+    if len(pos)==0:
+        ## the empty graph 
+        return {}
+    pos_v = np.array(list(pos.values()))
```

Comment:
The AttributeError you mention will occur if you use ```pos.values()``` without the ```np.array``` call.  There should be no AttributeError with ```np.array(pos.values())``` because the object is a numpy array which DOES have a ```shape``` attribute.  So you can take out the ```list``` here.

Also, you should not have to create new pull requests for each change. Simply commit the changes to the same branch you made the original changes to. The changes accumulate, but are "squashed" into a single commit when we merge the pull request into the repository. You can "push" changes to the same branch and they automatically get added to the pull request.  This is better than creating many pull requests because the entire history and comments, etc are stored in a single webpage for that pull request.

The test you created doesn't have any nodes in it. So, the test doesn't actually use any code inside a loop over nodes. A test with one node would do more, but it seems like you really need at least 2 nodes to test the rescale feature. Perhaps create a position dict with 2 nodes and integer values positions. Then set scale=2 and see if the positions double.